contrib/polygon/polygon.go

```
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/alpacahq/marketstore/contrib/polygon/api"
+	"github.com/alpacahq/marketstore/executor"
+	"github.com/alpacahq/marketstore/planner"
+	"github.com/alpacahq/marketstore/plugins/bgworker"
+	"github.com/alpacahq/marketstore/utils/io"
+	"github.com/golang/glog"
+	nats "github.com/nats-io/go-nats"
+)
+
+type PolygonFetcher struct {
+	config    FetcherConfig
+	backfillM sync.Map
+}
+
+type FetcherConfig struct {
+	// polygon API key for authenticating with their APIs
+	APIKey string `json:"api_key"`
+	// polygon API base URL in case it is being proxied
+	// (defaults to https://api.polygon.io/)
+	BaseURL string `json:"base_url"`
+}
+
+// NewBgWorker returns a new instances of PolygonFetcher. See FetcherConfig
+// for more details about configuring PolygonFetcher.
+func NewBgWorker(conf map[string]interface{}) (bgworker.BgWorker, error) {
+	data, _ := json.Marshal(conf)
```

Could ignoring this err ever go wrong? I wasn't sure.

---

This part is just to cast the same content from map to the concrete struct.  According to godoc, the only error case from `json.Marshal()` is when channel and other unsupported types are in the object tree. I don't think that's the case here.